apps/shared/avifpng.c

```diff
@@ -396,11 +419,15 @@ avifBool avifPNGWrite(const char * outputFilename, const avifImage * avif, uint3
 
     volatile int rgbDepth = requestedDepth;
     if (rgbDepth == 0) {
-        if (avif->depth > 8) {
-            rgbDepth = 16;
-        } else {
+#if defined(AVIF_ENABLE_EXPERIMENTAL_YCGCO_R)
+        if (avif->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_YCGCO_RE) {
             rgbDepth = 8;
+        } else {
+            rgbDepth = (avif->depth > 8) ? 16 : 8;
         }
+#else
+        rgbDepth = (avif->depth > 8) ? 16 : 8;
+#endif
```

Comment:
This is missing error checking. I think lines 421-431 should read something like the following:

```
    if (rgbDepth == 0) {
        rgbDepth = (avif->depth > 8) ? 16 : 8;
    }
#if defined(AVIF_ENABLE_EXPERIMENTAL_YCGCO_R)
    if (avif->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_YCGCO_RO) {
        fprintf(stderr, "AVIF_MATRIX_COEFFICIENTS_YCGCO_RO cannot be used with PNG because it has an even bit depth.\n");
        goto cleanup;
    }
    if (avif->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_YCGCO_RE) {
        if (avif->depth != 10 || (requestedDepth && requestedDepth != 8)) {
            fprintf(stderr, "...error message...\n");
            goto cleanup;
        }
        rgbDepth = 8;
    }   
#endif
```

Reply:
Done